packages/php-wasm/universal/src/lib/php-request-handler.ts

```diff
@@ -492,6 +498,58 @@ export class PHPRequestHandler<PHP extends BasePHP> {
 		}
 		throw new Error(`File not found: ${resolvedFsPath}`);
 	}
+
+	/**
+	 * Add paths to static files we can assume exist remotely.
+	 *
+	 * @param relativePaths A list of paths to remote assets, relative to the document root.
+	 */
+	addRemoteAssetPaths(relativePaths: string[]) {
+		const separator = this.#DOCROOT.endsWith('/') ? '' : '/';
+		relativePaths.forEach((relativePath) => {
+			const fsPath = `${this.#DOCROOT}${separator}${relativePath}`;
+			this.#remoteAssetPaths.add(fsPath);
+		});
+	}
+
+	/**
+	 * Guesses whether the given path looks like a PHP file.
+	 *
+	 * @example
+	 * ```js
+	 * seemsLikeAPHPRequestHandlerPath('/index.php') // true
+	 * seemsLikeAPHPRequestHandlerPath('/index.php') // true
+	 * seemsLikeAPHPRequestHandlerPath('/index.php/foo/bar') // true
+	 * seemsLikeAPHPRequestHandlerPath('/index.html') // false
+	 * seemsLikeAPHPRequestHandlerPath('/index.html/foo/bar') // false
+	 * seemsLikeAPHPRequestHandlerPath('/') // true
+	 * ```
+	 *
+	 * @param  path The path to check.
+	 * @returns Whether the path seems like a PHP server path.
+	 */
+	async #seemsLikeAPHPRequestHandlerPath(path: string): Promise<boolean> {
+		if (
+			this.#seemsLikeAPHPFile(path) ||
+			this.#seemsLikeADirectoryRoot(path)
+		) {
+			return true;
+		}
+
+		const primaryPhp = await this.getPrimaryPhp();
+		const fileDoesNotExist = !primaryPhp.fileExists(path);
+		const isNotAKnownRemoteAsset = !this.#remoteAssetPaths.has(path);
+		return fileDoesNotExist && isNotAKnownRemoteAsset;
+	}
+
+	#seemsLikeAPHPFile(path: string) {
+		return path.endsWith('.php') || path.includes('.php/');
```

Comment:
Should `.php/` also check for endsWith? This would prevent paths like `/test.php/index.js` from being executed.

```suggestion
		return path.endsWith('.php') || path.endsWith('.php/');
```

Reply:
Yeah, that check is wrong, but I was leaving it for a follow up PR because it is fixing a different problem from this PR. Originally, I'd actually removed that check altogether, which is also the wrong thing to do.

I have a note to make a different PR for this and will make sure the `something.php/` case is tested.

Reply:
> I have a note to make a different PR for this and will make sure the something.php/ case is tested.

Nevermind, we can go ahead and [refactor everything here](https://github.com/WordPress/wordpress-playground/pull/1417#discussion_r1604515501).